### PR TITLE
JR2-827: Check blackout expiration

### PR DESCRIPTION
**Pull request by birom**

### CODE REVIEW

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisementEvent.java [hunk 1]
```diff
@@ -14,8 +13,8 @@ public JobAdvertisementEvent(JobAdvertisementEvents jobAdvertisementEventType, J
     }
 
     @Override
-    public String getAggregateId() {
-        return this.jobAdvertisementId.getValue();
+    public JobAdvertisementId getAggregateId() {
+        return this.jobAdvertisementId;
     }
 
     public JobAdvertisementId getJobAdvertisementId() {
```
[the remark this answers is not in the document]
Comment (birom) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisementEvent.java]: True, I remove it.

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisementEvent.java [hunk 2]
```diff
@@ -1,7 +1,6 @@
 package ch.admin.seco.jobs.services.jobadservice.domain.jobadvertisement;
 
 import ch.admin.seco.jobs.services.jobadservice.core.domain.events.DomainEvent;
-import ch.admin.seco.jobs.services.jobadservice.core.domain.events.DomainEventType;
 
 public class JobAdvertisementEvent extends DomainEvent {
```
[the remark this answers is not in the document]
Comment (birom) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisementEvent.java]: No, only the JobAdvertisementEvents is used.

[the remark this answers is not in the document]
Comment (cjpsycos) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisementEvent.java]: i mean DomainEvent<JobAdvertisementId>

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java [hunk 3]
```diff
@@ -121,12 +124,20 @@ public void refining(JobAdvertisementId jobAdvertisementId) {
...
+
+    }
+
     public void publish(JobAdvertisementId jobAdvertisementId) {
         Condition.notNull(jobAdvertisementId, "JobAdvertisementId should not be null");
         JobAdvertisement jobAdvertisement = getJobAdvertisement(jobAdvertisementId);
         if (jobAdvertisement.isReportingObligation() && REFINING.equals(jobAdvertisement.getStatus())) {
             jobAdvertisement.publishRestricted();
-        } else {
+        } else if (PUBLISHED_RESTRICTED.equals(jobAdvertisement.getStatus())) {
```
[the remark this answers is not in the document]
Comment (birom) [reply application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java]: fixed

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisementEvent.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (birom) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisementEvent.java]: Thanks, fixed

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisementEvent.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (birom) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisementEvent.java]: fixed

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisement.java [hunk 4]
```diff
@@ -396,6 +358,14 @@ public void approve(String stellennummerAvam, LocalDate date, boolean reportingO
         DomainEventPublisher.publish(new JobAdvertisementEvent(JobAdvertisementEvents.JOB_ADVERTISEMENT_APPROVED, this));
     }
 
+    public void expireBlackout() {
+        Condition.notNull(reportingObligationEndDate, "Reporting obligation end date is missing");
+        Condition.isTrue(this.reportingObligationEndDate.isBefore(LocalDate.now()), "Must not be published before reportingObligationEndDate.");
```
[the remark this answers is not in the document]
Comment (birom) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisement.java]: fixed

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisement.java [hunk 5]
```diff
@@ -553,4 +523,358 @@ private boolean applyUpdates(JobAdvertisementUpdater updater) {
...
+            this.languageSkills = languageSkills;
+            return this;
+        }
+
+        public JobAdvertisementBuilder professionCodes(List<String> professionCodes) {
+            this.professionCodes = professionCodes;
+            return this;
+        }
+
+        public JobAdvertisement build() {
```
[the remark this answers is not in the document]
Comment (birom) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobAdvertisement.java]: fixed

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JabAdvertisementEventListener.java [hunk 6]
```diff
@@ -27,7 +27,7 @@ void onCreated(JobAdvertisementEvent jobAdvertisementEvent) {
         if (!JOB_ADVERTISEMENT_CREATED.getDomainEventType().equals(jobAdvertisementEvent.getDomainEventType())) {
```
Comment (cjpsycos) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JabAdvertisementEventListener.java]: can you please rename the class name from jab to job

Comment (birom) [reply application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JabAdvertisementEventListener.java]: Sure! it is done.

**Pull request merged**